xilem_web/src/after_update.rs

```diff
@@ -1,55 +1,107 @@
 // Copyright 2023 the Xilem Authors
 // SPDX-License-Identifier: Apache-2.0
 
-use xilem_core::{MessageResult, Mut, View, ViewId, ViewPathTracker};
+use xilem_core::{MessageResult, Mut, View, ViewId, ViewMarker};
 
-use crate::{interfaces::Element, DynMessage, ViewCtx};
+use crate::{DomView, DynMessage, ViewCtx};
 
-pub struct AfterUpdate<E, F> {
-    pub(crate) element: E,
-    pub(crate) callback: F,
+pub struct AfterBuild<E, F> {
+    element: E,
+    callback: F,
 }
 
-impl<E, F> AfterUpdate<E, F> {
-    pub fn new(element: E, callback: F) -> AfterUpdate<E, F> {
-        AfterUpdate { element, callback }
-    }
+pub struct AfterRebuild<E, F> {
+    element: E,
+    callback: F,
 }
 
-pub struct AfterUpdateState<E, S> {
+pub struct BeforeTeardown<E, F> {
     element: E,
-    child_state: S,
-    child_id: ViewId,
+    callback: F,
+}
+
+impl<E, F> AfterBuild<E, F> {
+    pub fn new(element: E, callback: F) -> AfterBuild<E, F> {
+        Self { element, callback }
+    }
+}
+
+impl<E, F> AfterRebuild<E, F> {
+    pub fn new(element: E, callback: F) -> AfterRebuild<E, F> {
+        Self { element, callback }
+    }
+}
+
+impl<E, F> BeforeTeardown<E, F> {
+    pub fn new(element: E, callback: F) -> BeforeTeardown<E, F> {
+        Self { element, callback }
+    }
+}
```

Comment:
We [leaned towards](https://github.com/linebender/xilem/pull/451#discussion_r1694948958) using functions directly for constructing views, so maybe rather use `after_build`, `after_rebuild` and `before_teardown`.

I also feel like we should document these, to explain, what this is about, and why these views shouldn't be used ^^.